Tests/XCUITests/TopTabsTest.swift

```diff
@@ -269,55 +269,54 @@ class TopTabsTest: BaseTestCase {
-//            XCTAssertTrue(app.cells[ImageIdentifiers.newTab].exists)
-//            XCTAssertTrue(app.cells["tab_close"].exists)
-//
-//            // Open New Tab
-//            app.cells[ImageIdentifiers.newTab].tap()
-//            navigator.performAction(Action.CloseURLBarOpen)
-//
-//            waitForTabsButton()
-//            checkNumberOfTabsExpectedToBeOpen(expectedNumberOfTabsOpen: 2)
-//            waitForExistence(app.cells.staticTexts["Home"])
-//            app.cells.staticTexts["Home"].firstMatch.tap()
-//
-//            // Close tab
-//            navigator.nowAt(HomePanelsScreen)
-//            navigator.performAction(Action.CloseURLBarOpen)
-//            navigator.nowAt(NewTabScreen)
-//
-//            waitForExistence(app.buttons["Show Tabs"])
-//            app.buttons["Show Tabs"].press(forDuration: 1)
-//            waitForExistence(app.cells[ImageIdentifiers.newTab])
-//            app.cells["tab_close"].tap()
-//            navigator.performAction(Action.CloseURLBarOpen)
-//            navigator.nowAt(NewTabScreen)
-//            checkNumberOfTabsExpectedToBeOpen(expectedNumberOfTabsOpen: 1)
-//
-//            // Go to Private Mode
-//            waitForExistence(app.cells.staticTexts["Home"])
-//            app.cells.staticTexts["Home"].firstMatch.tap()
-//            navigator.nowAt(HomePanelsScreen)
-//            navigator.performAction(Action.CloseURLBarOpen)
-//            navigator.nowAt(NewTabScreen)
-//            waitForExistence(app.buttons["Show Tabs"])
-//            app.buttons["Show Tabs"].press(forDuration: 1)
-//            waitForExistence(app.cells["nav-tabcounter"])
-//            app.cells["nav-tabcounter"].tap()
-//            navigator.performAction(Action.CloseURLBarOpen)
-//            navigator.nowAt(NewTabScreen)
-//            checkNumberOfTabsExpectedToBeOpen(expectedNumberOfTabsOpen: 1)
-//        }
+    func testLongTapTabCounter() {
+        if !iPad() {
+            waitForExistence(app.buttons["urlBar-cancel"], timeout: 5)
+            // Long tap on Tab Counter should show the correct options
+            navigator.performAction(Action.CloseURLBarOpen)
+            navigator.nowAt(NewTabScreen)
+            waitForExistence(app.buttons["Show Tabs"], timeout: 10)
+            app.buttons["Show Tabs"].press(forDuration: 1)
+            waitForExistence(app.cells.otherElements[ImageIdentifiers.newTab])
+            XCTAssertTrue(app.cells.otherElements[ImageIdentifiers.newTab].exists)
+            XCTAssertTrue(app.cells.otherElements["tab_close"].exists)
+
+            // Open New Tab
+            app.cells.otherElements[ImageIdentifiers.newTab].tap()
+            navigator.performAction(Action.CloseURLBarOpen)
+
+            waitForTabsButton()
+            checkNumberOfTabsExpectedToBeOpen(expectedNumberOfTabsOpen: 2)
+            waitForExistence(app.cells.staticTexts["Homepage"])
+            app.cells.staticTexts["Homepage"].firstMatch.tap()
+
```

Comment:
```suggestion
waitForExistence(app.collectionViews.cells["TopSitesCell"])
```
Adding this line can ensure that the screen is loaded before attempting to close the keyboard.